legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/TestCompilerHelper.java

```
@@ -12,52 +12,66 @@
             }
-
+            else if (mappingTest instanceof MappingDataTest)
+            {
+                MappingDataTest mappingDataTest = (MappingDataTest) mappingTest;
+                Root_meta_pure_mapping_metamodel_MappingDataTest_Impl compiledQueryTest = new Root_meta_pure_mapping_metamodel_MappingDataTest_Impl("");
+                compiledQueryTest
+                    ._storeTestData(ListIterate.collect(mappingDataTest.storeTestData, ele -> HelperMappingBuilder.processMappingElementTestData(ele, context, new ProcessingContext("Mapping Element: "))));
+                compiledMappingTest = compiledQueryTest;
+            }
+            else
+            {
+                throw new EngineException("Unsupported Mapping Test", test.sourceInformation, EngineErrorType.COMPILATION);
+            }
+            compiledMappingTest._id(mappingTest.id);
             List<String> assertionIds = ListIterate.collect(mappingTest.assertions, a -> a.id);
             List<String> duplicateAssertionIds = assertionIds.stream().filter(e -> Collections.frequency(assertionIds, e) > 1).distinct().collect(Collectors.toList());
-
             if (!duplicateAssertionIds.isEmpty())
             {
                 throw new EngineException("Multiple assertions found with ids : '" + String.join(",", duplicateAssertionIds) + "'", mappingTest.sourceInformation, EngineErrorType.COMPILATION);
             }
-            pureMappingTest._assertions(ListIterate.collect(mappingTest.assertions, assertion -> assertion.accept(new TestAssertionFirstPassBuilder(context, processingContext))));
-            return pureMappingTest;
-        }
-        else
-        {
-            return null;
+            compiledMappingTest._assertions(ListIterate.collect(mappingTest.assertions, assertion -> assertion.accept(new TestAssertionFirstPassBuilder(context, processingContext))));
+            // post checks
+            if (mappingTest.assertions == null || mappingTest.assertions.isEmpty())
```

should we throw this error before itself?